Content.Server/Zombies/ZombieSystem.cs

```diff
@@ -88,19 +111,43 @@ private void OnMeleeHit(EntityUid uid, ZombieComponent component, MeleeHitEvent
                 if (HasComp<ZombieComponent>(entity))
                     args.BonusDamage = -args.BaseDamage * zombieComp.OtherZombieDamageCoefficient;
 
-                if ((mobState.IsDead() || mobState.IsCritical())
+                if ((mobState.CurrentState == DamageState.Dead || mobState.CurrentState == DamageState.Critical)
                     && !HasComp<ZombieComponent>(entity))
                 {
                     _zombify.ZombifyEntity(entity);
                     args.BonusDamage = -args.BaseDamage;
                 }
-                else if (mobState.IsAlive()) //heals when zombies bite live entities
+                else if (mobState.CurrentState == DamageState.Alive) //heals when zombies bite live entities
                 {
                     var healingSolution = new Solution();
                     healingSolution.AddReagent("Bicaridine", 1.00); //if OP, reduce/change chem
                     _bloodstream.TryAddToChemicals(args.User, healingSolution);
                 }
             }
         }
+
+        public override void Update(float frameTime)
+        {
+            base.Update(frameTime);
+
+            foreach (var zombiecomp in EntityQuery<ZombieComponent>())
+            {
```

Comment:
You should have some kind of "ActiveZombieComponent" that gets tracked instead as it's just for the groan timer. Right now it's going to iterate every zombie every tick, dead or alive, so will lead to unnecessary iteration.